core/src/main/scala/chisel3/internal/Error.scala

```diff
@@ -5,7 +5,34 @@ package chisel3.internal
 import scala.annotation.tailrec
 import scala.collection.mutable.{ArrayBuffer, LinkedHashMap}
 
-class ChiselException(message: String, cause: Throwable = null) extends Exception(message, cause) {
+object ExceptionHelpers {
+
+  implicit class ThrowableHelpers(a: Throwable) {
+
+    /** Package names whose stack trace elements should be trimmed when generating a trimmed stack trace */
+    private val blacklistPackages: Set[String] = Set("chisel3", "scala", "java", "jdk", "sun", "sbt")
+
+    /** The object name of Chisel's internal `Builder`. Everything stack trace element after this will be trimmed. */
+    private val builderName: String = chisel3.internal.Builder.getClass.getName
+
+    def trimStackTraceToUserCode(blacklist: Set[String] = blacklistPackages, anchor: String = builderName): Unit = {
```

Comment:
```suggestion
    /** Package names whose stack trace elements should be trimmed when generating a trimmed stack trace */
    private val trimlistPackages: Set[String] = Set("chisel3", "scala", "java", "jdk", "sun", "sbt")

    /** The object name of Chisel's internal `Builder`. Everything stack trace element after this will be trimmed. */
    private val builderName: Option[String] = Some(chisel3.internal.Builder.getClass.getName)

    def trimStackTraceToUserCode(trimlist: Set[String] = trimlistPackages, anchor: Option[String] = builderName): Unit = {
```
Minor terminology suggestion and maybe making the `anchor` optional since it's possible one might not want an anchor.

Reply:
Good idea. Fixed in the latest commit.